addon/components/bs-button.js

```diff
@@ -436,39 +376,39 @@ export default class Button extends Component {
    */
   @action
   handleClick(e) {
-    let onClick = this.onClick;
-    let preventConcurrency = this.preventConcurrency;
+    let onClick = this.args.onClick;
+    let preventConcurrency = this.args.preventConcurrency;
 
-    if (onClick === null || onClick === undefined) {
+    if (!onClick) {
       return;
     }
 
     if (!preventConcurrency || !this.isPending) {
-      let promise = onClick(this.value);
+      let promise = onClick(this.args.value);
```

Comment:
Yeah, so the ButtonGroup's version of a button needs a value, as that determines the value of the group (e.g. an array of all selected button values). The base button does not really need a value, I agree. I think we can refactor that eventually, but only separately as a breaking change!

Reply:
I'm still thinking that this is better done baby curling the arguments using `{{fn}}` helper.

For the `<Button group::Button>` it's not that obvious cause how it extends from `<BsButton>`. Instead of invoking the `<BsButton>` component in a template it extends from its class.  To be honest I don't like that pattern.

I think the `<BsButtonGroup::Button>` should be refactored to something like this:

```hbs
<BsButton
  @active={{this.isActive}}
  @onClick={{fn @onClick @value}}
  @type={{@type}}
  @buttonType={{@buttonType}}
  ... attributes
  as |button|
>
  {{yield button}}
</BsButton>
```

Reply:
Don't get me wrong. This is nothing which is blocking this PR. I will see if I have some time in the next days to provide a PR showcasing such a refactoring.